models/RegistrationChecks.php

```diff
@@ -135,11 +135,28 @@ public function showAgeCheck()
 
         /** @var Module $module */
         $module = Yii::$app->getModule('legal');
-        if ($module->showAgeCheck()) {
-            if ($this->user === null || empty($module->settings->user($this->user)->get(static::SETTING_KEY_AGE))) {
+
+        // Check if the user and their profile property are not null before accessing them
+        if ($module->showAgeCheck() && $this->user !== null && $this->user->profile !== null) {
+            // Calculate the user's age
+            $birthday = $this->user->profile->birthday;
+            $minimumAge = $module->getMinimumAge();
+            $currentYear = date('Y');
+            $userBirthYear = date('Y', strtotime($birthday));
+            $userAge = $currentYear - $userBirthYear;
+
+            // Check if the user's age meets the minimum age requirement
+            if ($userAge < $minimumAge) {
+                // Perform logout if the age check fails
+                Yii::$app->user->logout();
```

Comment:
I am unsure whether we should simply log out here, e.g. a user could have entered an incorrect date of birth. We would then log them out without further notice.

Reply:
We could attempt a redirect or display a error message requesting to contact an Administrator for more information? The problem is the birthday field can always be manipulated when it comes to adding a date because the user must click the checkbox field which they see the age beforehand no?